lib/trino-hdfs/src/main/java/io/trino/filesystem/hdfs/HdfsFileIterator.java

```diff
@@ -68,11 +68,14 @@ public FileEntry next()
         }
 
         String root = listingPath.toUri().getPath();
+        if (!root.endsWith("/")) {
+            root += "/";
+        }
         String path = status.getPath().toUri().getPath();
```

Comment:
`status.getPath().toUri()` -- does it do normalization when path contains repeated slashes?
do we have test covering file listings from locations with double slashes?